connectors/serializers/base.py

```diff
@@ -0,0 +1,11 @@
+from abc import abstractmethod
+from typing import List, Tuple
+
+
+class BaseSerializer:
+    def __init__(self, output_path):
+        self.output_path = output_path
+    
+    @abstractmethod
+    def serialize(self, data: Tuple[List[str]], columns: List[str]):
```

Comment:
Maybe the type of `data` should be
```suggestion
    def serialize(self, data: Iterable[Iterable[Any]], columns: List[str]):
```
because we probably don't want to assume it's strings everywhere? Also, `Iterable`s would allow streaming data into the serializer.